raiden-network/light-client
# Bugfix/dapp connect token screen shows transfer header

**Thank you for submitting this pull request :)**

Fixes #2675

**Short description**

Solution description with background details in the linked issue. All other infos in the according commits. 
This includes some nice improvements of the router guards.

**Definition of Done**


**Steps to manually test the change (dApp)**

1. Connect to the dApp and use an account you never used for it before
2. Make sure you see the route to connect your frist token
3. Try to access the transfer route for example and make sure you get redirected again
4. Make sure you can still access all account routes (e.g. to transfer ETH or deposit to UDC)
5. Add your first token/open a channel and make sure all routes are accessible for that
6. Make sure you finally can access the transfer route (also after reload/re-connect)
